refactor!: redesign minimizer plugin options (#7444)
web-infra-dev/rspack: website/docs/en/plugins/rspack/lightning-css-minimizer-rspack-plugin.mdx

## Options

### include

- **Type:** `string | RegExp | (string | RegExp)[]`
- **Default:** `undefined`

> Use this to specify which files should be minified.

Reviewer: Make it more clear:

```suggestion
Use this to specify which files should be minified, it matches the path of the output files.
```

### exclude

- **Type:** `string | RegExp | (string | RegExp)[]`
- **Default:** `undefined`

> Use this to specify which files should be excluded from minification.

Reviewer: ```suggestion
Use this to specify which files should be excluded from minification, it matches the path of the output files.
```

### test

- **Type:** `string | RegExp | (string | RegExp)[]`
- **Default:** `undefined`

> Use this to provide a pattern that css files are matched against. If the filename matches the given pattern, it will be minified, otherwise it won't be.

Reviewer: ```suggestion
Use this to provide a pattern that CSS files are matched against. If the output filename matches the given pattern, it will be minified, otherwise it won't be.
```

.a{color: red}
```

### minimizerOptions

Configuration passed to Lightning CSS for minification.

> Below are the configurations supported. For detailed configuration, please refer to [lightningcss document](https://lightningcss.dev/transpilation.html)

Reviewer: ```suggestion
Below are the configurations supported. For detailed usage, please refer to [Lightning CSS documentation](https://lightningcss.dev/transpilation.html)
```

```ts
type LightningcssFeatureOptions = {
  nesting?: boolean;
  notSelectorList?: boolean;
  dirSelector?: boolean;
  langSelectorList?: boolean;
  isSelector?: boolean;
  textDecorationThicknessPercent?: boolean;
  mediaIntervalSyntax?: boolean;
  mediaRangeSyntax?: boolean;
  customMediaQueries?: boolean;
  clampFunction?: boolean;
  colorFunction?: boolean;
  oklabColors?: boolean;
  labColors?: boolean;
  p3Colors?: boolean;
  hexAlphaColors?: boolean;
  spaceSeparatedColorNotation?: boolean;
  fontFamilySystemUi?: boolean;
  doublePositionGradients?: boolean;
  vendorPrefixes?: boolean;
  logicalProperties?: boolean;
  selectors?: boolean;
  mediaQueries?: boolean;
  color?: boolean;
};

> type LightningCssMinimizerOptions = {

Reviewer: We should tell users that all features are excluded by default, and recommend that users to configure `targets`
